indra/llui/llscrolllistctrl.cpp

```diff
@@ -1105,38 +1105,32 @@ S32 LLScrollListCtrl::getItemIndex( const LLUUID& target_id ) const
 
 void LLScrollListCtrl::selectPrevItem( bool extend_selection)
 {
+    updateSort();
+
     LLScrollListItem* prev_item = NULL;
 
-    if (!getFirstSelected())
-    {
-        // select last item
-        selectNthItem(getItemCount() - 1);
-    }
-    else
+    for (LLScrollListItem* item : mItemList)
     {
-        updateSort();
-
-        item_list::iterator iter;
-        for (LLScrollListItem* cur_item : mItemList)
+        if (item->getSelected())
         {
-            if (cur_item->getSelected())
-            {
-                if (prev_item)
-                {
-                    selectItem(prev_item, cur_item->getSelectedCell(), !extend_selection);
-                }
-                else
-                {
-                    reportInvalidInput();
-                }
-                break;
-            }
+            break;
+        }
 
-            // don't allow navigation to disabled elements
-            prev_item = cur_item->getEnabled() ? cur_item : prev_item;
+        // don't allow navigation to disabled elements
+        if (item->getEnabled())
+        {
+            prev_item = item;
         }
     }
 
+    if (!prev_item)
+    {
+        reportInvalidInput();
```

Comment:
This seem like a change in behavior.
Previously if nothing was selected it would select last item, which is expected, but now if nothing was selected it would reportInvalidInput();?